models/object_detection_yolox/YoloX.py

```diff
@@ -0,0 +1,125 @@
+        return predictions
+
+
+    def postprocess(self, outputs, ratio):
+        grids = []
+        expanded_strides = []
+        hsizes = [self.input_size[0] // stride for stride in self.strides]
+        wsizes = [self.input_size[1] // stride for stride in self.strides]
+
+        for hsize, wsize, stride in zip(hsizes, wsizes, self.strides):
+            xv, yv = np.meshgrid(np.arange(hsize), np.arange(wsize))
+            grid = np.stack((xv, yv), 2).reshape(1, -1, 2)
+            grids.append(grid)
+            shape = grid.shape[:2]
+            expanded_strides.append(np.full((*shape, 1), stride))
+
+        grids = np.concatenate(grids, 1)
+        expanded_strides = np.concatenate(expanded_strides, 1)
+        outputs[..., :2] = (outputs[..., :2] + grids) * expanded_strides
+        outputs[..., 2:4] = np.exp(outputs[..., 2:4]) * expanded_strides
+
+        predictions = outputs[0]
+
+        boxes = predictions[:, :4]
+        scores = predictions[:, 4:5] * predictions[:, 5:]
+
+        boxes_xyxy = np.ones_like(boxes)
+        boxes_xyxy[:, 0] = boxes[:, 0] - boxes[:, 2] / 2.
+        boxes_xyxy[:, 1] = boxes[:, 1] - boxes[:, 3] / 2.
+        boxes_xyxy[:, 2] = boxes[:, 0] + boxes[:, 2] / 2.
+        boxes_xyxy[:, 3] = boxes[:, 1] + boxes[:, 3] / 2.
+        boxes_xyxy /= ratio
+
+        final_dets = []
+        num_classes = scores.shape[1]
+        for cls_ind in range(num_classes):
+            cls_scores = scores[:, cls_ind]
+            valid_score_mask = cls_scores > self.confThreshold
+
+            if valid_score_mask.sum() == 0:
+                continue
+
+            else:
+                valid_scores = cls_scores[valid_score_mask]
+                valid_boxes = boxes_xyxy[valid_score_mask]
+
+                keep = []
+                x1 = valid_boxes[:, 0]
+                y1 = valid_boxes[:, 1]
+                x2 = valid_boxes[:, 2]
+                y2 = valid_boxes[:, 3]
+
+                areas = (x2 - x1 + 1) * (y2 - y1 + 1)
+                order = valid_scores.argsort()[::-1]
+
+                while order.size > 0:
+                    i = order[0]
+                    keep.append(i)
+                    xx1 = np.maximum(x1[i], x1[order[1:]])
+                    yy1 = np.maximum(y1[i], y1[order[1:]])
+                    xx2 = np.minimum(x2[i], x2[order[1:]])
+                    yy2 = np.minimum(y2[i], y2[order[1:]])
+
+                    w = np.maximum(0.0, xx2 - xx1 + 1)
+                    h = np.maximum(0.0, yy2 - yy1 + 1)
+                    inter = w * h
+                    ovr = inter / (areas[i] + areas[order[1:]] - inter)
+
+                    inds = np.where(ovr <= self.nmsThreshold)[0]
+                    order = order[inds + 1]
+                    if len(keep) > 0:
+                        cls_inds = np.ones((len(keep), 1)) * cls_ind
+                        dets = np.concatenate([valid_boxes[keep], valid_scores[keep, None], cls_inds], 1)
+                        final_dets.append(dets)
```

Comment:
call `cv2.dnn.NMSBoxes()` to perform NMS. Documentation for `NMSBoxes()`: https://docs.opencv.org/4.x/d6/d0f/group__dnn.html#ga9d118d70a1659af729d01b10233213ee